### PR TITLE
Remove Agent override version in integration tests

### PR DESCRIPTION
**Pull request by pchila**
We used to override the agent version to a previous version between the version bump for a new release and the availability of the first snapshot build (usually a period of a few days) because Fleet had a limitation about not accepting agent versions more recent than itself.

This limitation should have been removed by now so we can get rid of the agent version override and use the default agent version

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by jlind23 — APPROVED** [review]

**Comment by jlind23** [comment]
Buildkite test it

File: .buildkite/scripts/steps/integration_tests.sh [hunk 1]
```diff
@@ -8,22 +8,13 @@ MAGE_TARGET="${2:-"integration:test"}"
 MAGE_SUBTARGET="${3:-""}"
 
 
-# Override the agent package version using a string with format <major>.<minor>.<patch>
-# There is a time when the snapshot is not built yet, so we cannot use the latest version automatically
-# This file is managed by an automation (mage integration:UpdateAgentPackageVersion) that check if the snapshot is ready.
-OVERRIDE_AGENT_PACKAGE_VERSION="$(cat .package-version)"
```
Comment (pierrehilbert) [inline .buildkite/scripts/steps/integration_tests.sh:14]: I don't think we should stop doing that!
We need https://github.com/elastic/elastic-agent/pull/5100 to be merged instead.

**Review by pierrehilbert — CHANGES_REQUESTED** [review]
We shouldn't remove all those parts according to me.

**Comment by pchila** [comment]
@pierrehilbert This PR came after a discussion in the elastic-agent control plane weekly meeting about #5096 where we determined that Fleet should be able to handle a version of elastic-agent newer than itself.

If Fleet is able to manage a newer agent we don't need to package elastic-agent with a previous version after the version bump. If we still need to use an older stack version on cloud that should be accomplished by specifying `AGENT_STACK_VERSION` when running the tests... Maybe that is the reason why you say we still need this code ?

/cc @cmacknz @ycombinator

**Comment by pierrehilbert** [comment]
Yes indeed my objective here is to ensure we can have our tests green even if the new stack version is not available on our cloud.
But if you have another idea, I'm ofc opened to it.

**Comment by pchila** [comment]
@pierrehilbert  what you is probably what is in commit 83a3a830df4dcea100c13b33f3fea06003ba5daf
Please have another look

**Review by pierrehilbert — APPROVED** [review]
Thanks for switching AGENT_VERSION to AGENT_STACK_VERSION as this is what we will need in the future.

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**